addons/sourcemod/scripting/freak_fortress_2.sp

```diff
@@ -6555,274 +6546,149 @@ stock GetBossIndex(client)
 
 stock ParseFormula(client, const String:key[], const String:defaultFormula[], defaultValue)
 {
-	decl String:formula[1024], String:buffer[2];
-	new String:strValue[1024], Float:sum[128], calcOperators:_operator[128];
-	new bool:closeBracket, bool:usePlayers, bool:foundOperator, bracket;
+	decl String:formula[1024], String:bossName[1024];
 	KvRewind(BossKV[Special[client]]);
+	KvGetString(BossKV[Special[client]], "name", bossName, sizeof(bossName), "");
 	KvGetString(BossKV[Special[client]], key, formula, sizeof(formula), defaultFormula);
 	ReplaceString(formula, sizeof(formula), " ", ""); //Get rid of spaces
-	new length=strlen(formula);
-	for(new i; i<=length; i++)
+	new result=RoundFloat(IterateFormula(formula, key, bossName));
+	Debug("result = %i", result);
+	if(result<=0)
 	{
-		strcopy(buffer, sizeof(buffer), formula[i]);
-		Debug("buffer[0] contains %c", buffer[0]);
-		switch (buffer[0])
+		Debug("MALFORMATION!");
+		LogError("[FF2] %s has a malformed %s formula, using default!", bossName, key);
+		return defaultValue;
+	}
+	if(bMedieval)
+	{
+		return RoundFloat(result/3.6); //TODO: Make this configurable
+	}
+	return result;
+}
+
+enum Operators {
+	Operator_None = -1,
+	Operator_Add,
+	Operator_Subtract,
+	Operator_Multiply,
+	Operator_Divide,
+	Operator_Exponent,
+};
+
+stock Operate(Handle:sum, bracket, Float:v2, Handle:_operator)
+{
+	new Float:oSum = GetArrayCell(sum, bracket);
+	switch (GetArrayCell(_operator, bracket))
+	{
+		case Operator_Add:
 		{
-			case '\0':
+			SetArrayCell(sum, bracket, oSum + v2);
+		}
+		case Operator_Subtract:
+		{
+			SetArrayCell(sum, bracket, oSum - v2);
+		}
+		case Operator_Multiply:
+		{
+			SetArrayCell(sum, bracket, oSum * v2);
+		}
+		case Operator_Divide:
+		{
+			if (FloatCompare(v2, 0.000001) == -1 || FloatCompare(v2, -0.000001) == 1)
 			{
-				foundOperator = true; //Probably less hacky way of detecting constant hp.
-				Debug("Null terminator found!");
+				LogError("[FF2] Cowardly refusing divide by zero!");
+				return;
 			}
+			SetArrayCell(sum, bracket, oSum / v2);
+		}
+		case Operator_Exponent:
+		{
+			SetArrayCell(sum, bracket, Pow(oSum, v2));
+		}
+		default:
+		{
+			SetArrayCell(sum, bracket, v2);
+		}
+	}
+	SetArrayCell(_operator, bracket, Operator_None);
+}
+
+stock OperateString(Handle:sum, bracket, String:strValue[], size, Handle:_operator)
+{
+	if (StrEqual(strValue, ""))
```

Comment:
We can save a line here by inverting StrEqual and getting rid of the return; statement